modules/storages/app/common/storages/peripherals/oauth_configurations/one_drive_configuration.rb

```diff
@@ -52,6 +51,20 @@ def authorization_state_check(access_token)
           end
         end
 
+        def extract_origin_user_id(rack_access_token)
+          Net::HTTP.start(@uri.host, @uri.port, use_ssl: true) do |http|
+            response = http.get('/v1.0/me',
+                                { Authorization: "Bearer #{rack_access_token.access_token}", Accept: 'application/json' })
+
+            # if response == Net::HTTPSuccess
+            parsed = JSON.parse(response.body)
+            parsed['userPrincipalName']
```

Comment:
@mereghost `userPrincipalName` is an email address, right? Is that the best unique identifier that we should use? I mean, people change their names and their company domain names. And we need that info to create shares for automatically managed project folders. Isn't it more stable to use the user's `id`, instead? // @dominic-braeunlein, @Kharonus

Reply:
Hmmm... the UPN is a mix of o both (email + business domain) that will eventually be dealt by the AD to figure out who is who. I've avoided the user id as, when we add/remove permissions sending an object id we can open the doors to allowing a group by accident.

That being said, it is a minor change, so let's try it.

Reply:
Aren't there email addresses for groups, too? Then both attributes could reference a group and both attributes have the same "issue".

Reply:
Oh, groups there was a reference for a security group, rather than a mail distribution group.